File: 19-03-20-done/Q1.py
# -*- coding:utf-8 -*-
class Solution:
    def multiply(self, A):
        # write code here
        B = []
        n = len(A)
        for i in range(n):
            s = 1
            for j in range(n):
                if (j != i):
                    s = s * A[j]
            B.append(s)
        return B

File: 19-03-20/Q1.cpp
class Solution {
public:
    vector<int> multiply(const vector<int>& A) {
        vector<int> B(A.size(), 1);
        for(int i = 1; i < A.size(); i++){
            B[i] = A[i-1] * B[i-1];
        }
        int temp = 1;
        for(int i = A.size() - 1; i >= 0; i--){
            B[i] *= temp;
            temp *= A[i];
        }
        return B;
    }
};
